walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,180 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.ListIterator;
 
 public final class FindMeetingQuery {
+  private int meetingDuration;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // The total (common) free time we have to conduct the meeting.
+    LinkedList<TimeRange> freeTime = new LinkedList<TimeRange>();
+    freeTime.addAll(Arrays.asList(TimeRange.WHOLE_DAY));
+
+    meetingDuration = (int) request.getDuration();
+    LinkedList<String> optionalAttendees = new LinkedList<>(request.getOptionalAttendees());
```

Comment:
Why do you create a dedicated list for optionalAttendees but use the mandatory attendees in the request as they are? Also why use a LinkedList for attendees? It does make sense for the TimeRanges, but I don't see a point in using this datastructure for the attendees.

Reply:
I stored the optional attendees as a dedicated list cause I am passing and using it from 12 different places where as mandatory attendees in the request are only used twice.

Reply:
Yes I see, I now changed the variable type to 'Collection', that the MeetingRequest class is returning. (Inside which is an HashSet object). And also created a variable for mandatoryAttendees of the same type.